cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -279,11 +279,6 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 	defer func() {
 		validateResult(osbuildJobResult, job.Id().String())
```

Comment:
I think that's similar to the comment from @thozza. I started thinking of how to clean this up, but may possibly delay it to a follow-up is it looks quite fragile/hairy. Or do you think this PR is making it worse?

Reply:
Slightly, because `osbuildJobResult` is now returned. Previously, everything happened in this function. Anyway, I think it's fine to leave it as is. Getting it clean feels out of scope of this PR.